sdk/src/Services/S3/Custom/Util/AmazonS3Uri.cs

```diff
@@ -34,6 +34,10 @@ public class AmazonS3Uri
     {
         private const string EndpointPattern = @"^(.+\.)?s3[.-]([a-z0-9-]+)\.";
 
+        private static Regex endpointRegexMatch;
+
+        private static Regex EndpointRegexMatch => endpointRegexMatch ?? (endpointRegexMatch = new Regex(EndpointPattern, RegexOptions.Compiled));
```

Comment:
Also update L80 to use `EndpongRegexMatch`

Reply:
Good catch. I'll update.